packages/Angular/explorer-core/src/lib/dashboard-browser-component/dashboard-browser.component.ts

```diff
@@ -68,35 +187,109 @@ export class DashboardBrowserComponent {
     const result = await dashboardEntity.Save();
     if(result){
       this.dashboards.push(dashboardEntity);
+
+      let item: Item<DashboardEntity> = new Item(dashboardEntity, ItemType.Resource);
+      this.items.push(item);
     }
     else{
       this.sharedService.CreateSimpleNotification("Error creating new dashboard entity", "error");
     }
     this.showLoader = false;
   }
 
-  public async deleteItem(item: DashboardEntity): Promise<void> {
-    this.showLoader = true;
-    if (item && item.ID && item.ID > 0) {
+  public async deleteItem(item: Item<DashboardEntity | Folder>): Promise<void> {
```

Comment:
@JS-BC think about ways we could push this functionality (and others like it), generically, into the lower level base/generic component. One idea would be to handle deletes (and other CRUD ops) in the base component by specifying the entity for the item type and the entity for the folders and then the internal component uses that to do the work itself by getting data, running views and so forth. You'll likely find as you implement other similar wrapper components to this one that 90% of the code for Delete and other methods is basically the same. As you do that, think about pushing this functionality INSIDE the generic component and then exposing events that you can use to extend/override the functionality as well.